schematics/src/keys-manager/index.ts

```diff
@@ -1,17 +1,20 @@
-import { Rule, Tree, SchematicsException } from '@angular-devkit/schematics';
+import { Rule, Tree, SchematicsException, SchematicContext } from '@angular-devkit/schematics';
 import { TranslocoConfig } from '@ngneat/transloco-utils';
-import { execSync } from 'child_process';
 import { addScriptToPackageJson } from '../utils/package';
 import { getWorkspace, setWorkspace } from '../utils/projects';
 import { getConfig, updateConfig } from '../utils/transloco';
 import { SchemaOptions } from './schema';
 
-function installKeysManager() {
-  console.log('Installing packages for tooling...');
-  execSync('npm install --save-dev @ngneat/transloco-keys-manager ngx-build-plus');
+import { addPackageJsonDependency, NodeDependencyType } from '@schematics/angular/utility/dependencies';
+import { NodePackageInstallTask } from '@angular-devkit/schematics/tasks';
+import { KEYS_MANAGER_VERSION, NGX_BUILD_PLUS_VERSION } from '../schematics.consts';
+
+function installDependencies(context: SchematicContext) {
+  context.logger.log('info', `🔍 Installing packages...`);
+  context.addTask(new NodePackageInstallTask());

```

Comment:
maybe a better solution would be figuring out if the project is using yarn or npm and simply run the correct command?

Reply:
That's probably too much work for a schematic? The tooling for adding packages and executing the package manager for the project already exist as provided from the angular team.

Reply:
> maybe a better solution would be figuring out if the project is using yarn or npm and simply run the correct command?

And assuming the package manager is not going to be 100% accurate all the time.

Reply:
The reason we didn't use this build-in tool is that we needed to specify the version instead of installing the latest version, I would look how angular decide to use yarn or npm and use the same method.

Reply:
> The reason we didn't use this build-in tool is that we needed to specify the version instead of installing the latest version and as far as I know, Angular's API does not support that, 

The angular API installs a specified version.

Reply:
yap, so that's why it isn't good enough for us :)

Reply:
If you put this script into the keys-manager repo, this problem will go away.

>  we needed to specify the version instead of installing the latest version and as far as I know, Angular's API does not support that


I think you meant you "needed to install the latest version instead of specifying the version.."